docs/ssh.md

```diff
@@ -0,0 +1,68 @@
+# KubeOne and SSH
```

Comment:
I'd probably name this something along `KubeOne SSH requirements` or `How KubeOne uses SSH`.

Reply:
👍